### DISCUSSION TITLE
tensorflow dependencies for Guesslang vs go-enry

### DISCUSSION THREAD

**Discussion by ccoVeille** [robherley/snips.sh#182] [Ideas]
You are using Guesslang so you require using tensorflow. This is pretty huge dependencies, something out of go ecosystem.

Did you try https://github.com/go-enry/go-enry ?

**Comment by robherley** [comment]
👋 Sorry bit late on this, but I've moved to use magika model instead which uses onnx: 
- https://github.com/robherley/snips.sh/pull/251

Unfortunately a lot of libraries (like `go-enry` that you shared) utilize other metadata like file extension which we don't have at upload time. Since it's just static text, we require the use of an model to infer.

This is not yet released (will do so with a few other enhancements) but the latest tag should be updated.
